tableau_utilities/scripts/connection.py

```diff
@@ -0,0 +1,112 @@
+    conn_warehouse = args.conn_warehouse
+
+    if args.conn == 'settings_yaml':
+        conn_user = settings['embed_connection']['username']
+        conn_pw = settings['embed_connection']['password']
+        conn_type = settings['embed_connection']['class_name']
+        conn_db = settings['embed_connection']['dbname']
+        conn_schema = settings['embed_connection']['schema']
+        conn_host = settings['embed_connection']['server']
+        conn_role = settings['embed_connection']['service']
+        conn_warehouse = settings['embed_connection']['warehouse']
+    elif args.conn == 'os_env':
+        conn_user = os.getenv("CONN_USERNAME")
+        conn_pw = os.getenv("CONN_PASSWORD")
+        conn_type = os.getenv("CONN_TYPE")
+        conn_db = os.getenv("CONN_DB")
+        conn_schema = os.getenv("CONN_SCHEMA")
+        conn_host = os.getenv("CONN_HOST")
+        conn_role = os.getenv("CONN_ROLE")
+        conn_warehouse = os.getenv("CONN_WAREHOUSE")
+
+
+    connection_settings = {'conn_user': conn_user,
+                           'conn_pw': conn_pw,
+                           'conn_type': conn_type,
+                           'conn_db': conn_db,
+                           'conn_schema': conn_schema,
+                           'conn_host': conn_host,
+                           'conn_role': conn_role,
+                           'conn_warehouse': conn_warehouse
+                           }
+
+    return connection_settings
+
+
+def update_connection(datasource, conn_settings):
+    """
+
+    Args:
+        datasource: The datasource object
+        conn_settings: The connection setttings dictionary
+
+    """
+    connection_type = conn_settings['conn_type']
+    named_conn = datasource.connection.named_connections.get(conn_settings['conn_type'])
+
+    # named_conn = conn_settings['conn_type']
+    print(named_conn)
+
+    named_conn.connection.class_name = connection_type
+    named_conn.connection.server = conn_settings['conn_host']
+    named_conn.connection.username = conn_settings['conn_user']
+    named_conn.connection.service = conn_settings['conn_role']
+    named_conn.connection.dbname = conn_settings['conn_db']
+    named_conn.connection.schema = conn_settings['conn_schema']
+    named_conn.connection.warehouse = conn_settings['conn_warehouse']
+
+    datasource.connection.named_connections[connection_type].connection = named_conn
+    print(named_conn)
+    datasource.save()
+    print('saved the datasource')
+
+
+def connection(args, server=None):
+    """
+
+    """
+
+    print("in the connection file")
+
+
+
+    # Set the connection settings dictionary
+    conn_settings = connection_settings(args)
+
+    # Get the datasource
+    if args.datasource_source == 'local':
+        print('Local Datasource')
+        datasource = tu.Datasource(args.datasource_path)
+
+    elif args.datasource_source == 'online':
+        print('Online Datsource')
+
+    update_connection(datasource, conn_settings)
```

Comment:
@JustinGrilli  I'm running into a recursion error when I try to get the replacing connection in a local file to work.  

Do you have any idea how to fix this?  I'm giving up for a day or two for now.
```
Traceback (most recent call last):
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/bin/tableau_utilities", line 33, in <module>
    sys.exit(load_entry_point('tableau-utilities==2.0.0', 'console_scripts', 'tableau_utilities')())
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/scripts/cli.py", line 299, in main
    args.func(args)
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/scripts/connection.py", line 106, in connection
    update_connection(datasource, conn_settings)
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/scripts/connection.py", line 82, in update_connection
    datasource.save()
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file.py", line 211, in save
    parent.insert(starting_index, section.xml())
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 72, in xml
    return ET.fromstring(xmltodict.unparse({self.tag: self.dict()}, pretty=True))
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 829, in dict
    dictionary['named-connections'] = {'named-connection': [nc.dict() for nc in self.named_connections]}
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 829, in <listcomp>
    dictionary['named-connections'] = {'named-connection': [nc.dict() for nc in self.named_connections]}
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 651, in dict
    'connection': self.connection.dict()
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 651, in dict
    'connection': self.connection.dict()
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 651, in dict
    'connection': self.connection.dict()
  [Previous line repeated 989 more times]
RecursionError: maximum recursion depth exceeded
```

Here's my command for trying to run this:
```
 op run --env-file="/Users/jayrosenthal/.test_env" -- tableau_utilities --auth os_env  --datasource_source local --datasource_path '/Users/jayrosenthal/code/tableau-utilities/tmp_tdsx_and_config/Opportunity Field History.tdsx'  connection --conn os_env
```

You should be able to try it with something like:
```
tableau_utilities --auth settings_yaml .... <ADD_THE_ARG_FOR_YOUR_FILE>  --datasource_source local --datasource_path '/Users/jayrosenthal/code/tableau-utilities/tmp_tdsx_and_config/Opportunity Field History.tdsx'  connection --conn os_env

Reply:
Okay, so I think this is an issue of you assigning the `NamedConnection` to the `NamedConnection.connection`, on line `80`; Rather than assigning the `NamedConnection.connection`.

I am going to add a suggestion above related to this :)

Reply:
Cool!  Thanks for helping!